### PR TITLE
Update sink-to-elasticsearch.md

### LINKED ISSUE #1693: Change required: Fix Es jsonb doc
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

Now, our es sink will send JSON as a JSON string, and ES will convert it into a struct structure and write it into it. So need to update docs.

jsonb  ~~text~~  =>  JSON string，ES will automatically convert it into a struct.
We need to make sure that the structure type in the json is consistent to prevent es from reporting errors

### Reference

_No response_

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ Add a note to remind users of a point that might get less data in Elaticsearch.]
 

- **Related doc issue**
  
  Resolves [https://github.com/risingwavelabs/risingwave-docs/issues/1693#event-11367913247]

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by ShanlanLi** [comment]
@xxhZs For 

"Now, our es sink will send JSON as a JSON string, and ES will convert it into a struct structure and write it into it. So need to update docs.

jsonb text => JSON string，ES will automatically convert it into a struct.", 

My update in the documentation is:

image

I'm unsure if it's right, please help check it.

File: docs/guides/sink-to-elasticsearch.md [hunk 1]
```diff
@@ -85,3 +85,10 @@ ElasticSearch uses a mechanism called [dynamic field mapping](https://www.elasti
 |struct |Not supported|
 |array |array|
 |JSONB |text|
+|json |struct (RisingWave's ElasticSearch sink will send JSON as a JSON string, and ElasticSearch will convert it into a struct structure)|
```
Comment (xxhZs) [inline docs/guides/sink-to-elasticsearch.md]: remove json, 
and |jsonb |struct (RisingWave's ElasticSearch sink will send JSONB as a JSON string, and ElasticSearch will convert it into a struct structure)|

And we support struct now

File: docs/guides/sink-to-elasticsearch.md [hunk 2]
```diff
@@ -24,11 +24,11 @@ The Elasticsearch sink connector in RisingWave provides at-least-once delivery s
...
+- If you are running RisingWave locally from binaries, make sure that you have [JDK 11](https://openjdk.org/projects/jdk/11/) or later versions installed in your environment.
 
-## Create a Elasticsearch sink
+## Create an Elasticsearch sink
 
-Use the following syntax to create a Elasticsearch sink. Once a sink is created, any insert or update to the sink will be streamed to the specified Elasticsearch endpoint.
+Use the following syntax to create an Elasticsearch sink. Once a sink is created, any insert or update to the sink will be streamed to the specified Elasticsearch endpoint.
 
 ```sql
 CREATE SINK sink_name
```
Comment (xxhZs) [inline docs/guides/sink-to-elasticsearch.md:34]: Please also fix a bug here, remove type

File: docs/guides/sink-to-elasticsearch.md [hunk 3]
```diff
@@ -82,6 +82,12 @@ ElasticSearch uses a mechanism called [dynamic field mapping](https://www.elasti
 |timestamp without time zone | text|
 |timestamp with time zone |text|
 |interval |text|
-|struct |Not supported|
+|struct |struct|
 |array |array|
-|JSONB |text|
+|JSONB|struct (RisingWave's ElasticSearch sink will send JSONB as a JSON string, and ElasticSearch will convert it into a struct structure)|
```
Comment (CharlieSYH) [inline docs/guides/sink-to-elasticsearch.md:87]:
```suggestion
|JSONB|struct (RisingWave's Elasticsearch sink will send JSONB as a JSON string, and ElasticSearch will convert it into a struct structure)|
```

File: docs/guides/sink-to-elasticsearch.md [hunk 4]
```diff
@@ -82,6 +82,12 @@ ElasticSearch uses a mechanism called [dynamic field mapping](https://www.elasti
...
+
+:::note
+Unlike normal databases, Elasticsearch doesn't require users to explicitly `CREATE TABLE`. Instead, it infers the schema on-the-fly based on the first record ingested. For example, if a record contains a jsonb '{v1: 100}', v1 will be inferred as a long type. However, if the next record is '{v1: "abc"}', the ingestion will fail because "abc" is inferred as a string and the two types are incompatible.
```
Comment (CharlieSYH) [inline docs/guides/sink-to-elasticsearch.md:90]: I suggest we remove "Unlike normal databases,"

**Review by CharlieSYH — APPROVED** [review]

File: docs/guides/sink-to-elasticsearch.md (hunk 3 above)
Comment (xuefengze) [reply docs/guides/sink-to-elasticsearch.md:87]: `struct/JSONB` is `object` type? 
https://www.elastic.co/guide/en/elasticsearch/reference/current/object.html

File: docs/guides/sink-to-elasticsearch.md (hunk 1 above)
Comment (xuefengze) [reply docs/guides/sink-to-elasticsearch.md]: `struct/JSONB` is `object` type?
https://www.elastic.co/guide/en/elasticsearch/reference/current/object.html

**Comment by ShanlanLi** [comment]
Because there are several prs which all edit this file the same time, to avoid conflicts, I copy all the changes to one of them. This issue would be resolved by https://github.com/risingwavelabs/risingwave-docs/pull/1760.

**Pull request closed without merging**